keps/sig-node/runtime-class-scheduling.md

```diff
@@ -190,43 +181,26 @@ inculde:
   RuntimeClasses would need a toleration to enable them to be run on those
   nodes.
 
-### RuntimeController
+### Scheduler
 
-RuntimeController is a new in-tree admission plugin that should eventually be
-enabled on almost every Kubernetes clusters. The role of the controller for
-scheduling is to merge the topology constraints from the RuntimeClass into the
-PodSpec. Eventually, the controller's responsibilities may grow, such as to
-merge in [pod overhead][] or validate feature compatibility.
+A new scheduler predicate will manage the RuntimeClass topology. It will lookup
+the RuntimeClass associated with the pod being scheduled. If there is no
+RuntimeClass, or the RuntimeClass does not include a topology, then the
+predicate will permit the pod to be scheduled to the evaluated node. Otherwise,
+it will check whether the NodeSelector matches the node.
 
-Note that the RuntimeController is not needed if we implement [native scheduler
-support](#runtimeclass-aware-scheduling).
-
-#### Mix-in
-
-The RuntimeClass topology is merged with the pod's NodeSelector & Tolerations.
+The `PodToleratesNodeTaints` predicate will also be extended to be
```

Comment:
Can't we just call from the `RuntimeClassPredicate` the `PodToleratesNodeTaints` with the right parameters ? composing seems easier than modifying each predicate to include runtimeClass awareness

Reply:
I assume predicates are AND'd together, in which case the PodToleratesNodeTaints predicate would also be called without the runtimeclass tolerations, and in that instance fail? I'm not very familiar with the scheduler code though, so I could be totally wrong.

Reply:
This raises a broader Issue at predicates level, when a predicate is basically a composition of other:

- do we add domain awareness to the existing predicates ?
- do we compose these predicates into domain predicate ?

This might over complicate our predicates definition, and create performance overhead in some cases.